test/functional/CacheManagement/src/tests/sharedclasses/options/TestSharedCacheJavaAPI.java

```diff
@@ -66,8 +88,41 @@ public static void main(String args[]) {
 		    if (oldCacheCount == -1) {
 		    	fail("iterateSharedCacheFunction failed");
 		    }
-		    persistentCount = nonpersistentCount = snapshotCount = 0;
+	    	
+	    	if (dir == null && false == isWindows() && false == isMVS() && isOpenJ9()) {
+	    		dirGroupAccess = getCacheDir("Foo_groupaccess", false);
+	    		dirRemoveJavaSharedResources = removeJavaSharedResourcesDir(dirGroupAccess);
+	    		cacheGroupAccessList = SharedClassUtilities.getSharedCacheInfo(dirGroupAccess, SharedClassUtilities.NO_FLAGS, false);
+		    	if (cacheGroupAccessList == null) {
+		    		oldCacheGroupAccessCount = 0;
+		    	} else {
+		    		oldCacheGroupAccessCount = cacheList.size();
+		    	}
+			    if (oldCacheGroupAccessCount == -1) {
+			    	fail("iterateSharedCacheFunction failed");
```

Comment:
Better to include the directory in the error message. But this code seems unreachable. Can cacheList.size() return -1 ? Similarly for the following cases.

Reply:
This is similar to original migrated code, which do check this case. Similar fuction iterateSharedCache(dir, NO_FLAGS, false) does return -1 
https://ci.adoptopenjdk.net/view/work%20in%20progress/job/Grinder_Sandbox/895/consoleFull

Reply:
Yes, original code was checking the return value of iterateSharedCache(), which can be -1. But looking at the current code here:

```
     if (cacheGroupAccessList == null) {
	oldCacheGroupAccessCount = 0;
     } else {
	oldCacheGroupAccessCount = cacheList.size();
     }
     if (oldCacheGroupAccessCount == -1) {
```
oldCacheGroupAccessCount is either 0 or cacheList.size(), neither of which can be -1.

Reply:
It is actually testing SharedClassUtilities.getSharedCacheInfo(), but the error message says "iterateSharedCacheFunction failed", which is confusing. iterateSharedCache() is being tested in class TestSharedCacheJvmtiAPI instead of this class.

Reply:
Updated. Thanks!